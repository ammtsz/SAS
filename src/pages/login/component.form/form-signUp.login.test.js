import React from "react";
import SignUp from "./form-signUp.login.component";

import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

import {
  actionSetAuthError,
  actionEmailSignUp,
} from "../../../redux/user/user.actions.js";

describe("Component login-page <SignUp />", () => {
  const mockStore = configureMockStore();
  const store = mockStore({
    user: {
      datas: null,
      error: null,
    },
  });
  store.dispatch = jest.fn();

  const mockNewAccount = jest.fn();
  const mockProps = {
    newAccount: mockNewAccount,
  };

  let component;
  beforeEach(() => {
    component = render(
      <Provider store={store}>
        <MemoryRouter>
          <SignUp {...mockProps} />
        </MemoryRouter>
      </Provider>
    );
  });

  afterEach(() => {
    cleanup();
  });

  it("should render SignUp", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

  it("should contain 'Sign Up' text", () => {
    expect(screen.getByText("Sign Up")).toBeInTheDocument();
  });

  it("should contain name, email, password and confirm-password inputs", () => {
    expect(screen.getByPlaceholderText("name")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("email")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("password")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("confirm password")).toBeInTheDocument();
  });

  it("should NOT contain Persistence button", () => {
    expect(screen.queryByTestId("signin-persistence")).toEqual(null);
  });

  it("should contain 'Create account' and 'Log in' buttons", () => {
    expect(screen.getByText("Create account")).toBeInTheDocument();
    expect(screen.getByText("Log in")).toBeInTheDocument();
  });

  it("should fire 'actionSetAuthError' if passwords don't match", () => {
    const displayName = "name";
    const email = "email@email.com";
    const password = "12345";
    const confirmPassword = "1234";

    const nameInput = screen.getByPlaceholderText("name");
    fireEvent.change(nameInput, { target: { value: displayName } });

    const emailInput = screen.getByPlaceholderText("email");
    fireEvent.change(emailInput, { target: { value: email } });

    const passwordInput = screen.getByPlaceholderText("password");
    fireEvent.change(passwordInput, { target: { value: password } });

    const confirmPasswordInput = screen.getByPlaceholderText(
      "confirm password"
    );
    fireEvent.change(confirmPasswordInput, {
      target: { value: confirmPassword },
    });

    expect(nameInput).toHaveValue(displayName);
    expect(emailInput).toHaveValue(email);
    expect(passwordInput).toHaveValue(password);
    expect(confirmPasswordInput).toHaveValue(confirmPassword);

    fireEvent.click(screen.getByText("Create account"));
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(
      actionSetAuthError("passwords don't match")
    );
  });

  it("should fire 'handleSubmit' when click on 'Create account' button", () => {
    const displayName = "name";
    const email = "email@email.com";
    const password = "12345";
    const confirmPassword = password;

    const nameInput = screen.getByPlaceholderText("name");
    fireEvent.change(nameInput, { target: { value: displayName } });

    const emailInput = screen.getByPlaceholderText("email");
    fireEvent.change(emailInput, { target: { value: email } });

    const passwordInput = screen.getByPlaceholderText("password");
    fireEvent.change(passwordInput, { target: { value: password } });

    const confirmPasswordInput = screen.getByPlaceholderText(
      "confirm password"
    );
    fireEvent.change(confirmPasswordInput, {
      target: { value: confirmPassword },
    });

    expect(nameInput).toHaveValue(displayName);
    expect(emailInput).toHaveValue(email);
    expect(passwordInput).toHaveValue(password);
    expect(confirmPasswordInput).toHaveValue(confirmPassword);

    fireEvent.click(screen.getByText("Create account"));
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(
      actionEmailSignUp({ email, password, displayName })
    );
  });

  it("should fire 'newAccount' when click on 'Log in' button", () => {
    fireEvent.click(screen.getByText("Log in"));
    expect(mockNewAccount).toHaveBeenCalledTimes(1);
  });
});
