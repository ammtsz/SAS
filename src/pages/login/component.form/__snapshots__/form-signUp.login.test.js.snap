// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component login-page <SignUp /> should render SignUp 1`] = `
<DocumentFragment>
  <h4
    class="sc-gKsewC fHGveF"
  >
    Sign Up
  </h4>
  <form
    class="sc-iBPRYJ jHiIsc"
  >
    <div
      class="sc-bdfBwQ kmCbvM form-input__box"
    >
      <label
        for="signup-name"
      >
        <i
          class="fas fa-user"
        />
      </label>
      <input
        class="sc-gsTCUz dDffFH form-input__input"
        id="signup-name"
        name="displayName"
        placeholder="name"
        type="text"
      />
    </div>
    <div
      class="sc-bdfBwQ kmCbvM form-input__box"
    >
      <label
        for="signup-email"
      >
        <i
          class="fas fa-envelope"
        />
      </label>
      <input
        class="sc-gsTCUz dDffFH form-input__input"
        id="signup-email"
        name="email"
        placeholder="email"
        type="email"
      />
    </div>
    <div
      class="sc-bdfBwQ kmCbvM form-input__box"
    >
      <label
        for="signup-password"
      >
        <i
          class="fas fa-lock"
        />
      </label>
      <input
        class="sc-gsTCUz dDffFH form-input__input"
        id="signup-password"
        name="password"
        placeholder="password"
        type="password"
      />
    </div>
    <div
      class="sc-bdfBwQ kmCbvM form-input__box"
    >
      <label
        for="signup-confirm-password"
      >
        <i
          class="fas fa-lock"
        />
      </label>
      <input
        class="sc-gsTCUz dDffFH form-input__input"
        id="signup-confirm-password"
        name="confirmPassword"
        placeholder="confirm password"
        type="password"
      />
    </div>
    <button
      class="sc-hKgILt sc-fubCfw ivrlBZ ywriB"
    >
      Create account
    </button>
    <button
      class="sc-hKgILt sc-fubCfw ivrlBZ ywriB"
    >
      Log in
    </button>
  </form>
</DocumentFragment>
`;
